syntax = "proto3";

package hadev;

message WriteRequest {
  int64 addr = 1;
  bytes data = 2;
}

message WriteReply {
  int32 ret = 1;
}

message ReadRequest {
  int64 addr = 1;
}

message ReadReply {
  int32 ret = 1;
  bytes data = 2;
}


service BlockStore {
  rpc Write (WriteRequest) returns (WriteReply) {}
  rpc Read (ReadRequest) returns (ReadReply) {}
}
